### PR TITLE
NMS-12283: Be able to execute ICMP requests through Java when running as non-root

### PR DESCRIPTION
**Pull request by indigo423**
### All Contributors

* [x] Have you read and followed our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [x] Have you [made an issue in the OpenNMS issue tracker](https://issues.opennms.org)?<br>If so, you should:
  1. update the title of this PR to be of the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
* [x] Have you made a comment in that issue which points back to this PR?
* [x] Have you updated the JIRA link at the bottom of this comment to link to your issue?
* [x] If this is a new or updated feature, is there documentation for the new behavior?
* [x] If this is new code, are there unit and/or integration tests?
* [x] If this PR targets a `foundation-*` branch, does it avoid changing files in `$OPENNMS_HOME/etc/`?

Added the network capabilities to the Java process similar as in the Horizon image to allow sending ICMP packets with Java. Additionally, the OpenJDK version is bumped to the latest available version 11.0.4.11 in the CentOS 7 repositories. We have also removed `gettext` and added `jq` which was a request to make things easier from @agalue. The 3 changes are separate commits so they can be reverted individually if necessary.

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-12283
* Bamboo (Continuous Integration): https://bamboo.opennms.org/

### CODE REVIEW

**Comment by agalue** [comment]
LGTM, thanks!

**Comment by agalue** [comment]
Although, the title should say "be able to perform ICMP through Java when running as non-root", because the non-root nature is already present.

**Comment by agalue** [comment]
Unfortunately, net.ipv4.ping_group_range requires using sysctl which only runs when you have root privileges. This is forbidden in strict environments like OpenShift.

Also, that flag works when the host (the machine that runs Docker/Kubelet) has a fairly recent kernel. In other words, that option won't work on RHEL/CentOS 7 or older. So, for OpenShift you would need CoreOS, which is the only option without going to OpenShift 4 with RHEL 8 to be able to use that privileged operation with OpenShift 3. Besides that, this operation has to be done by the sysadmins at OS level, as OpenShift itself won't allow you to do it through YAML or the UI for obvious reasons.

A similar situiation will happen with plain Docker, Swarm or Kubernets on restricted environments.

We have customers that are not allowed to run applications as root. While we help them deploying OpenNMS in their restricted environments, we found that the best and most elegant solution that doesn't impose kernel restrictions (in this customers case, only RHEL/CentOS 7 is allowed) to perform ICMP requests from within the JVM as a non-privileged user was using the netcap tool over the java binary.

In other words, the netcap solution for the java binary is well tested and verified with multiple customers regardless of how OpenNMS or Minion are running (containerized or bare-metal).

My $.02

**Comment by Naicisum** [comment]
Apologies, the echo command works on OpenShift when use in the Dockerfile and building the image. I have validated that this indeed does work on RHEL7 inside Openshift.

**Comment by indigo423** [comment]
Thank you @agalue and @Naicisum for taking the time to look into these changes! 🚀
